draft-ietf-pquip-pqc-hsm-constrained.md

```diff
@@ -406,47 +402,28 @@ and distribute malicious updates.
-
-* XMSS (eXtended Merkle Signature Scheme): Another stateful hash-based signature scheme similar to HSS-LMS
-  {{RFC8391}}. XMSS signatures are slightly shorter than HSS-LMS signatures for equivalent security. However, HSS-LMS provides performance advantages and HSS-LMS is considered
-  simpler (see Section 10 of {{RFC8554}}).
-
-Firmware images can be signed using one of these post-quantum algorithms before being
-distributed to constraied devices. {{?I-D.wiggers-hbs-state}} discusses various strategies
-for a correct state and backup management for stateful hash-based signatures.
-
-Firmware images often have a long lifetime, requiring cryptographic algorithms that
-provide strong security assurances over extended periods. ML-DSA is not included in this
-list because it is a lattice-based signature scheme, making it susceptible to potential
-advances in quantum and classical attacks on structured lattices. The long-term security
-of ML-DSA depends on the continued hardness of lattice-based problems, which remain an
-active area of research. In addition, since ML-DSA implementations are still maturing,
-relying on hash-based signatures can be a more reliable and production-ready option
-for firmware authentication where long-term cryptographic stability is critical.
-
-Hash-based signature schemes may be preferable to ML-DSA for firmware authentication,
-particularly in scenarios where long-term cryptographic stability is a critical
-requirement. Schemes such as SLH-DSA, HSS-LMS, and XMSS are built on well-understood
-hash functions, and their security does not rely on unproven assumptions like the
-hardness of lattice problems.
-
-While SLH-DSA benefits from being stateless and avoids the complexity of state management,
-its large signature sizes make it less suitable for memory-constrained devices. In contrast,
-HSS-LMS and XMSS offer significantly smaller signatures and can achieve efficient
-verification times, making them more practical choices for constrained environments
-where performance and memory usage are key concerns.
+To ensure the integrity and authenticity of firmware updates, constrained devices will have to adopt PQC digital signature schemes for code signing.
+These algorithms must provide long-term security, operate efficiently in low-resource environments, and be compatible with secure update mechanisms, such as the firmware update architecture for IoT described in {{!RFC9019}}.
+
+The Software Updates for Internet of Things (SUIT) working group is defining mandatory-to-implement cryptographic algorithms for IoT devices in {{?I-D.ietf-suit-mti}}, which recommends use of HSS-LMS {{RFC8554}} to secure software devices.
+
+Stateful hash-based signature schemes, such as HSS-LMS or the similar XMSS {{RFC8391}}, are good candidates for signing firmware updates.
+Their security is based on the security of the underlying hash function, which is well-understood.
```

Comment:
I would add:

```
Those schemes offer can achieve efficient verification times, making them more practical choices for constrained environments where performance and memory usage are key concerns.
```

Especially in case of LMS, the verification time is much faster than ML-DSA and SLH-DSA and this is very important for secure boot. I think this is the only scenario I would use LMS (after trying hard to avoid it).

BTW: SLH-DSA can be also fast (see https://eprint.iacr.org/2024/367)